### PR TITLE
Page & Empty States FAD-5745

### PR DESCRIPTION
**Pull request by jonambas**
All list pages look something like this:

```js
<Page
   primaryAction={{ content: 'Create Webhook', Component: Link, to: '/webhooks/create' }}
   title='Webhooks'
   empty={{
       test: webhooks.length === 0,
       image: 'Setup',
       title: 'Create a Webhook',
       content: <p>Push message events directly to your own endpoints</p>
   }}>
   { error ? this.renderError() : this.renderCollection() }
</Page>
```

### CODE REVIEW

File: src/pages/users/ListPage.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ import { Loading, ApiErrorBanner, DeleteModal, TableCollection } from 'src/compo
 import AccessSelect from './components/AccessSelect';
 import DeleteButton from './components/DeleteButton';
 
-const COLUMNS = ['Name', 'Role', 'Email', 'Last Login', null];
+const COLUMNS = ['Name', 'Role', 'Email', 'Last Login', ''];
```
Comment (jonambas) [inline src/pages/users/ListPage.js]: Think someone else fixed this in another branch, can maybe switch back to null

**Comment by jasonrhodes** [comment]
I don't have strong feelings about whether the "error" state is shown within the Page wrapper or happens outside like Loading. I do think that if we keep it within the page, we should either do a ternary for the fork, or move it out to some `this.renderPage` or `this.renderPageContent` method. The `{error &&` followed by `{!error &&` feels like it will cause us bugs down the road.

```jsx
{error ? this.renderError() : this.renderCollection()}
```
Ternary like that looks fine here, and shows that we intended to have an obvious if/else fork there.

**Comment by jonambas** [comment]
@jasonrhodes I can switch to a ternary - There are a couple places whit more than 2 render funcs:
Should I just combine "everything else shown without an error" into a single render function?

```js
<Page primaryAction={primaryAction} title="Users">
        { error && this.renderError() }
        { !error && this.renderCollection() }
        { !error && this.renderDeleteModal() }
</Page>
```

**Comment by jasonrhodes** [comment]
@jonambas yeah either `{error ? this.renderError() : this.renderNoError()}` or `{this.renderPageContent()}` and handle the error state with an early return in there.

File: src/pages/users/ListPage.js [hunk 2]
```diff
@@ -97,33 +95,39 @@ export class ListPage extends Component {
...
+  render() {
+    const { loading, error } = this.props;
+
+    if (loading) {
+      return <Loading />;
+    }
+
+    return (
+      <Page primaryAction={primaryAction} title="Users">
+        { error ? this.renderError() : this.renderPage() }
```
[the remark this answers is not in the document]
Comment (jonambas) [reply src/pages/users/ListPage.js]: I think your own account will always be shown, but we could add one

**Comment by jonambas** [comment]
I can change `test` in matchbox quickly, I think `show` would be clearer

**Pull request merged**